caPutLogApp/caPutLogTask.c

```diff
@@ -93,13 +93,14 @@ static void val_assign(VALUE *dst, const VALUE *src, short type);
 static void val_dump(LOGDATA *pdata);
 #endif
 
-static int shut_down = FALSE;           /* Shut down flag */
 static DBADDR caPutLogPV;               /* Structure to keep address of Log PV */
 static DBADDR *pcaPutLogPV;             /* Pointer to PV address structure,
                                            also used as a flag whether this
                                            PV is defined or not */
 static epicsMessageQueueId caPutLogQ;   /* Mailbox for caPutLogTask */
 
+static volatile int caPutLogConfig;
```

Comment:
You mean "instead of just marking it `volatile`"?
I am not sure if atomic is necessary. I think not much can go wrong it it isn't atomic.
The idea behind `volatile` was that the `while` loop in `caPutLogTask` really reads the variable each time. (Even though (most?) compilers handle global vars as volatile anyway, afaik.)

Reply:
Oops, yes I did mean instead of `volatile`. The IOC uses `volatile enum` control variables for similar task shutdowns in dbCa, dbScan and 3 instances in rsrv, but @mdavidsaver changed callback.c to use epicsAtomic operations a few years back, I guess they might give some protection on future architectures, not sure that it really matters.

Reply:
In the case of callback.c, I was trying to reduce the number of false positives flagged by valgrind/helgrind.

Reply:
So I stay with `volatile` for the moment until someone can show me how it fails.